soccer/gameplay/single_robot_sequence.py

```diff
@@ -0,0 +1,51 @@
+import behavior_sequence
+import single_robot_behavior
+import role_assignment
+
+#A sequence to be carried out by a single robot, attempting to add non-single robot behaviors will raise exceptons
+class SingleRobotSequence(behavior_sequence.BehaviorSequence, single_robot_behavior.SingleRobotBehavior):
+    def __init__(self, behaviors = [], continuos=False, autorestart=lambda: True):
+        single_robot_behavior.SingleRobotBehavior.__init__(self, continuos)
+        behavior_sequence.BehaviorSequence.__init__(self, behaviors = behaviors)
+ f
+    @property
+    def behaviors(self):
+        return behavior_sequence.BehaviorSequence.behaviors.fget(self)
+
+    @behaviors.setter
+    def behaviors(self, value):
+        #Insures that all behaviors are
+        for b in value:
+            if not isinstance(b, single_robot_behavior.SingleRobotBehavior):
+                raise TypeError("SingleRobotSequence requires single robot behaviors")
```

Comment:
If you subclass SingleRobotCompositeBehavior, I think you can add this type check to it's implementation of add_subbehavior() and avoid having to override BehaviorSequence's `behaviors` property.